frontend/src/concepts/pipelines/content/tables/pipelineRun/usePipelineRunFilter.ts

```diff
@@ -24,17 +26,25 @@ const usePipelineRunFilter = (
   runs: PipelineRunKF[],
 ): [filterJobs: PipelineRunKF[], toolbarProps: FilterProps] => {
   const [filterData, setFilterData] = React.useState<FilterData>(DEFAULT_FILTER_DATA);
+  const { getJobInformation } = useJobRelatedInformation();
 
   const filteredRuns = runs.filter((run) => {
     const runValue = filterData[FilterOptions.NAME];
     const experimentValue = filterData[FilterOptions.EXPERIMENT];
     const pipelineValue = filterData[FilterOptions.PIPELINE];
     const startedValue = filterData[FilterOptions.STARTED];
     const statusValue = filterData[FilterOptions.STATUS];
+    const { data } = getJobInformation(run);
```

Comment:
43 requests to the same endpoint:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/8126518/c5128699-fb14-47ea-a194-f84cce2f22fe)

This is a performance problem -- we need to centralize this information. Maybe we need a Context or something -- maybe adding it to the `usePipelineAPI` so it can reuse values is what we need to do here.